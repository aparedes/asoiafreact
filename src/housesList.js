/* @flow */
import * as Immutable from 'immutable'
import type { List, Map } from 'immutable'
import React, { Component } from 'react'
import type { Element } from 'react'
import House from './house'
import { connect } from 'react-redux'

type PropsType = {
  allHouses: List<string>,
  error: ?string,
  getting: boolean,
}

class HousesList extends Component {
  static displayName = 'HousesList'
  props: PropsType
  render(): Element<*> {
    const { allHouses, error, getting } = this.props
    return (
      <div>
        { getting ? 'Loading' : error }
        { allHouses.map((house) => <House key={ house } houseId={ house } /> ) }
      </div>
    )
  }
}

const mapStateToProps = (state: { houses: Map<string, *> }) => ({
  allHouses: state.houses.getIn([ 'housesIds' ], new Immutable.List()),
  error: state.houses.getIn([ 'error_all' ], null),
  getting: state.houses.getIn([ 'getting_all' ], false),
})

const mapDispatchToProps = (dispatch: (action: Object) => void): Object => ({ })

export default connect(mapStateToProps, mapDispatchToProps)(HousesList)
