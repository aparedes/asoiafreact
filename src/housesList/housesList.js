/* @flow */
import './houseList.css';
import * as Immutable from 'immutable';
import type { List, Map } from 'immutable';
import React, { Component } from 'react';
import type { Element } from 'react';
import HouseItem from './houseItem/houseItem';
import { connect } from 'react-redux';

type Props = {
  allHouses: List<string>,
  error: ?string,
  getting: boolean,
};

function mapHouses(house: string) {
  return <HouseItem key={house} houseId={house} />;
}

export function HousesList(props: Props) {
  const { allHouses, error, getting } = props;
  return (
    <div className={'houseList'}>
      {getting ? 'Loading' : error}
      {allHouses.map(mapHouses)}
    </div>
  );
}

type ReduxState = {
  app: { search: string, region: string },
  houses: Map<string, *>,
};
function mapStateToProps(state: ReduxState) {
  let allHouses;
  if (state.app.search && state.app.search.length > 0) {
    const search = new RegExp(state.app.search.toLowerCase());
    allHouses = state.houses
      .getIn(['houses'], new Immutable.Map())
      .filter(house => search.test(house.get('name', '').toLowerCase()))
      .sortBy(house => house.getIn(['name'], ''))
      .keySeq()
      .toList();
  } else if (state.app.region) {
    const region = new RegExp(state.app.region.toLowerCase());
    allHouses = state.houses
      .getIn(['houses'], new Immutable.Map())
      .filter(house => region.test(house.get('region', '').toLowerCase()))
      .sortBy(house => house.getIn(['name'], ''))
      .keySeq()
      .toList();
  } else {
    allHouses = state.houses.getIn(['housesIds'], new Immutable.List());
  }
  return {
    allHouses,
    error: state.houses.getIn(['error_all'], null),
    getting: state.houses.getIn(['getting_all'], false),
  };
}

export default connect(mapStateToProps)(HousesList);
