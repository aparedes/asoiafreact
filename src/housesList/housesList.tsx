import './houseList.css';
import React, { ReactElement } from 'react';
import HouseItem from './houseItem/houseItem';
import {  useSelector } from 'react-redux';
import { ReduxState } from '../store/constants/reduxTypes';


function mapHouses(house?: string): ReactElement<void> | null {
  if (!house) {
    return null;
  }
  return <HouseItem key={house} houseId={house} />;
}

export function HousesList() {
  const { allHouses, error, getting } = useSelector(mapStateToProps);

  return (
    <div className={'houseList'}>
      {getting ? 'Loading' : error}
      {allHouses.map(mapHouses)}
    </div>
  );
}

function mapStateToProps(state: ReduxState) {
  let allHouses;
  if (state.app.search && state.app.search.length > 0) {
    const search = new RegExp(state.app.search.toLowerCase());
    allHouses = state.houses.houses
      .filter((house: any) => search.test(house.name.toLowerCase()))
      .map((house) => house.id);
  } else if (state.app.region) {
    const region = new RegExp(state.app.region.toLowerCase());
    allHouses = state.houses.houses
      .filter((house) => region.test(house.region.toLowerCase()))
      .map((house) => house.id);
    // .sortBy((house: Map<string, unknown>) => house.getIn(['name'], ''))
  } else {
    allHouses = [...(state.houses.housesIds ?? [])];
  }
  console.warn(allHouses)
  return {
    allHouses,
    error: state.houses.errorAll,
    getting: state.houses.gettingAll,
  };
}

export default HousesList;
