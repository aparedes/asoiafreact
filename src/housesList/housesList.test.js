import * as Immutable from 'immutable'
import React from 'react';
import { HousesList } from './housesList'
import renderer from 'react-test-renderer'

jest.mock('../house/house', () => {
  const { House } = require.requireActual('../house/house')
  class FakeHouse extends React.Component {
  render() {
    return (
      <House
        houseId={ this.props.houseId }
        name={ 'House Stark of Winterfell' }
        coatOfArms={ 'Dire wolf' }
        word={ 'Winter is comming' }
      />
    )
  }
}
FakeHouse.propTypes = House.propTypes;
return FakeHouse
  return House
})
test('Prints a list of houses', () => {
  const component = renderer.create(
    <HousesList getting={ false } allHouses={ new Immutable.List([ '1' ])}/>
  );
  let housesList = component.toJSON();
  expect(housesList).toMatchSnapshot();
})
