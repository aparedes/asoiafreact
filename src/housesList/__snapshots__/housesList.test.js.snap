exports[`test Prints a list of houses 1`] = `
<div
  className="houseList">
  <div
    className="house"
    onClick={[Function]}>
    <div
      className="houseName">
      House Stark of Winterfell
    </div>
  </div>
</div>
`;
